src/auth/react-auth0-wrapper.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useEffect, useContext } from 'react'
+import createAuth0Client from '@auth0/auth0-spa-js'
+import routes from '../routes';
+
+const DEFAULT_REDIRECT_CALLBACK = () =>
+  window.history.replaceState({}, document.title, window.location.pathname);
+
+const auth0Options = {
+  domain: window.AUTH0_DOMAIN,
+  client_id: window.AUTH0_CLIENT_ID,
+  redirect_uri: `${window.location.origin}${routes.oauth}`
+}
+
+export const Auth0Context = React.createContext();
+export const useAuth0 = () => useContext(Auth0Context);
+export const Auth0Provider = ({
+  children,
+  onRedirectCallback = DEFAULT_REDIRECT_CALLBACK
+}) => {
+  const [isAuthenticated, setIsAuthenticated] = useState();
+  const [user, setUser] = useState();
+  const [auth0Client, setAuth0] = useState();
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const initAuth0 = async () => {
+      const auth0FromHook = await createAuth0Client(auth0Options);
+      setAuth0(auth0FromHook);
+
+      if (window.location.search.includes("code=")) {
+        const { appState } = await auth0FromHook.handleRedirectCallback();
+        onRedirectCallback(appState);
+      }
+
+      const isAuthenticated = await auth0FromHook.isAuthenticated();
+
+      setIsAuthenticated(isAuthenticated);
+
+      if (isAuthenticated) {
+        const user = await auth0FromHook.getUser();
+        setUser(user);
+      }
+
+      setLoading(false);
+    };
+    initAuth0();
+    // eslint-disable-next-line
```

Comment:
is this comment needed?

Reply:
That came in with the boilerplate provided by Auth0.  I'm sure we can remove it.